syntax = "proto3";
package ImageViewer;

message ZoomRequest {

	int64 send_start_time = 1; // the start time of sending the message
	int32 x_screensize_in_px = 2;
	int32 y_screensize_in_px = 3;
	int32 zoom_deltay = 4;

}

message ImageResponse {

	int64 task_start_time = 1;
	int64 send_start_time = 2;

	string filename = 3;
	
	message OneRow {
		repeated float row_data = 1;
	}
	repeated OneRow image_data = 4;
	
	int32 image_width = 5;
	int32 image_height = 6;
	int32 xmin = 7;
	int32 ymin = 8;
	float vmin = 9;
	float vmax = 10;
	
	float x_coor_min = 11;
	float x_coor_delta = 12;
	float y_coor_min = 13;
	float y_coor_delta = 14;
	float x_range_min = 15;
	float x_range_max = 16;
	float y_range_min = 17;
	float y_range_max = 18;

	float rebin_ratio = 19;

}
